src/internal/m365/collection/groups/collection.go

```diff
@@ -370,6 +378,44 @@ func (col *lazyFetchCollection[C, I]) streamItems(ctx context.Context, errs *fau
 				"item_id", id,
 				"parent_path", path.LoggableDir(col.LocationPath().String()))
 
+			// Handle metadata before data so that if metadata file fails,
+			// we are not left with an orphaned data file.
+			//
+			// If the data download fails for some reason other than deleted in
+			// flight, we will still end up persisting a .meta file. This is
+			// fine however, since the next backup will overwrite it.
+			//
+			// If item is deleted in flight, we will end up with an orphaned
+			// .meta file. The only impact here is storage bloat, which
+			// is minimal. Other impact could be if we do an in-order restore
+			// using a tree built from .meta files. We may have some .meta
+			// files without corresponding .data files.
+			itemMeta, _, err := col.getAndAugment.getItemMetadata(
+				ictx,
+				col.contains.container)
+			if err != nil && !errors.Is(err, errMetadataFilesNotSupported) {
+				errs.AddRecoverable(ctx, clues.StackWC(ctx, err))
+
+				return
+			}
+
+			if err == nil {
```

Comment:
Quick note, this code is slightly refactored in the following PR https://github.com/alcionai/corso/pull/5136/files .  Not doing it here to keep the PR within review limits.